bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/ConfigUtil.java

```diff
@@ -161,4 +171,130 @@ public static Object normalizeType(Object value, ConfigDescriptionParameter conf
         return convertedConfiguration;
     }
 
+    /**
+     * Return fields of the given class as well as all super classes.
+     *
+     * @param clazz The class
+     * @return A list of Field objects
+     */
+    public static List<Field> getAllFields(Class<?> clazz) {
```

Comment:
please make this `private` again.

Reply:
And it would be nice to have the public API on top and the supporting private methods at the bottom of the class.

Reply:
Yup, I usually would do that, but in this case all fields and methods of the ConfigUtil class are widely mixed. If I sort them now, it would be a hard job reviewing. Should be done in another PR, if that's ok.

Reply:
agreed, thats a pity.
But then please order at least the refactored code, so that public comes first, private comes last. And please make #getAllFields `private`.